src/pages/front-page.tsx

```diff
@@ -3,45 +3,49 @@ import React from 'react';
 import { Text } from '../components/text';
 import { Spinner } from '../components/spinner';
 import { Table, RowType } from '../components/table';
+import { InfiniteScroll } from '../components/infinite-scroll';
 
 import { useUsers } from '../graphql/queries/use-users';
 
 import { translations } from '../helpers/translations';
 
 const frontPageTranslations = translations.pt.front_page;
+const loadNumber = 10;
 
 export const FrontPage = () => {
+  const [rows, setRows] = React.useState<RowType[]>([]);
+  const [page, setPage] = React.useState(0);
   const [isLoading, setIsLoading] = React.useState(false);
 
   const header = ['Nome', 'Email'];
-  const [rows, setRows] = React.useState<RowType[]>([]);
 
   const loadUsers = async () => {
     setIsLoading(true);
-    const { data } = await useUsers();
-    const rows: RowType[] = data.users.nodes.map((user: RowType) => ({
+    const { data } = await useUsers(page);
```

Comment:
isso aqui pode gerar um problema, hooks não podem ser utilizados de forma condicional (se não me engano uma função é considerada condicional). Como eu coloquei no comentario anterior, essa `useUsers` não é um hook e pode não dar problema. Mas se fosse um hook isso aqui não poderia ser feito por causa da forma que o React trata os hooks ([link da documentação](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level))

Reply:
isso daria problema se eu utilizar `useQuery` (realmente uma hook) dentro do atual `useUser`?

Reply:
acredito que sim, o `useUser` passaria a ser um custom hook e entraria na regra de uso condicional

Reply:
no caso onde `useUser` é uma hook, como eu faria pra poder usar ele toda vez que ocorre o evento de fim de rolagem, definido pelo `InfiniteScroll`? Ou, como chamar ele com o prop `onBottomHit` se eu não posso colocar em uma função?